numba/tests/test_array_reductions.py

```diff
@@ -148,6 +154,18 @@ def test_argmax_basic(self):
         npr, nbr = run_comparative(array_argmax, arr)
         self.assertPreciseEqual(npr, nbr)
 
+    def test_median_odd(self):
+        arr = np.arange(101)
+        np.random.shuffle(arr)
```

Comment:
In addition to the random case, I think it would be nice to also test with some special cases: a sorted array, a reverse-sorted array, an array with all equal values.